lib/util/lifted/influx/httpd/handler_metrics.go

```diff
@@ -0,0 +1,246 @@
+package httpd
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/openGemini/openGemini/lib/metrics"
+	"github.com/openGemini/openGemini/lib/statisticsPusher/statistics"
+	"github.com/openGemini/openGemini/lib/syscontrol"
+	"github.com/openGemini/openGemini/lib/util/lifted/influx/influxql"
+	meta2 "github.com/openGemini/openGemini/lib/util/lifted/influx/meta"
+	"github.com/openGemini/openGemini/lib/util/lifted/influx/query"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.uber.org/zap"
+)
+
+var (
+	metricsHandler      http.Handler
+	openGeminiCollector *OpenGeminiCollector
+)
+
+func init() {
+	// 创建一个自定义的注册表
+	registry := prometheus.NewRegistry()
+	// 实例化自定义收集器
+	openGeminiCollector = NewOpenGeminiCollector()
+	// 注册收集器到注册表
+	err := registry.Register(openGeminiCollector)
+	if err != nil {
+		log.Fatal("Failed to register collector:", err)
+	}
+
+	metricsHandler = promhttp.HandlerFor(registry, promhttp.HandlerOpts{Registry: registry})
+}
+
+type OpenGeminiCollector struct {
+	*metrics.BaseCollector
+	indexMap map[string]*metrics.ModuleIndex
+}
+
+func NewOpenGeminiCollector() *OpenGeminiCollector {
+	c := &OpenGeminiCollector{
+		BaseCollector: metrics.NewBaseCollector(metrics.ComIndexRegistry),
+		indexMap:      make(map[string]*metrics.ModuleIndex),
+	}
+	// 初始化指标
```

Comment:
Be more international and use English annotations

Reply:
Okay, I've solved it.